cmake/FindFlyCapture2.cmake

```diff
@@ -84,67 +112,162 @@ set(FlyCapture2_FOUND TRUE)
 
 # Search supplied hint directories first if supplied.
 # Find include directory for flycapture
-find_path(FlyCapture2_INCLUDE_DIR
-    NAMES flycapture/FlyCapture2.h
-    PATHS ${FlyCapture2_INCLUDE_DIR_HINTS}
-    ${FlyCapture2_CHECK_INCLUDE_DIRS}
-    )
+if(WIN32)
+    #unfortunately Flycapture installs itself in an un-prepended include path in Windows
+    find_path(FlyCapture2_INCLUDE_DIR
+        NAMES FlyCapture2.h
+        PATHS ${FlyCapture2_INCLUDE_DIR_HINTS}
+        ${FlyCapture2_CHECK_INCLUDE_DIRS}
+        )
+else()
+    find_path(FlyCapture2_INCLUDE_DIR
+        NAMES flycapture/FlyCapture2.h
+        PATHS ${FlyCapture2_INCLUDE_DIR_HINTS}
+        ${FlyCapture2_CHECK_INCLUDE_DIRS}
+        )
+endif()
 if(NOT FlyCapture2_INCLUDE_DIR OR NOT EXISTS ${FlyCapture2_INCLUDE_DIR})
     FlyCapture2_REPORT_NOT_FOUND(
-        "Could not find flycapture include directory, set FlyCapture2_INCLUDE_DIR to "
-        "path to flycapture include directory,"
-        "e.g. /opt/flycapture.")
+    "Could not find flycapture include directory, set FlyCapture2_INCLUDE_DIR to "
+    "path to flycapture include directory,"
+    "e.g. /opt/flycapture.")
 else()
     message(STATUS "flycapture include dir found: " ${FlyCapture2_INCLUDE_DIR})
 endif()
 
+if(NOT WIN32)
 # Find library directory for flycapture
 find_library(FlyCapture2_LIBRARY
     NAMES libflycapture.so
     PATHS ${FlyCapture2_LIBRARY_DIR_HINTS}
     ${FlyCapture2_CHECK_LIBRARY_DIRS}
     )
+else()
+
+    if(MSVC)
```

Comment:
I think the `if(MSVC)` is only needed for the following block for setting `vTail` but not needed for the later `find_library` and `find_file`.